### ISSUE TITLE
Overlay Menu does not display in mobile preview when editing or viewing on iOS devices

### ISSUE THREAD

**Issue by mgozdis** [Automattic/wp-calypso#87158]
### Quick summary

Setting the overlay menu to Mobile only in various FSE themes, including `Hevor`, displays no overlay menu in mobile preview and no overlay menu at all on the frontend on iOS devices. Text links are displayed at all times regardless of the overlay menu setting on iOS devices. Reproducible on Simple and AT sites. Interaction: 7686104-zd-a8c

A workaround is deactivate Gutenberg. 

### Steps to reproduce

1. Create a new site
2. Activate Hevor theme (Not Hever)
3. Edit the Index template and check that the Navigation block has the overlay menu set to Mobile only
4. Change preview in Site Editor to Mobile preview and see the overlay menu does not display: ![e8yGDK.png](https://github.com/Automattic/wp-calypso/assets/121042930/d7aaf865-aa38-411d-830a-1165785aa20a)
5. View the live site on any iOS device in any browser and see that the overlay menu also does not display

### What you expected to happen

The overlay menu should display in mobile preview and on iOS devices when set to display on mobile.

### What actually happened

The overlay menu does not display in mobile preview or on iOS devices. It seems to display fine outside of the editor when scaling on desktop or checking on non-iOS mobile devices. 

### Impact

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Platform (Simple and/or Atomic)

Simple, Atomic

### Logs or notes

For iOS devices while testing on the frontend, we tested with an ancient iPhone 6 in Safari and Brave as well as an iPhone 13 Pro with Safari 16 and latest version of Chrome.

**Comment by filipanoscampos** [comment]
📌 REPRODUCTION RESULTS

**Replicated** partially on Simple and Atomic.
**Not replicated** on self-hosted.


📌 FINDINGS

The menu works on iOS for me, on my live test sites.
On my end, the issue is only affecting the live site.

📌 ACTIONS
– Requested author feedback
– Very similar to https://github.com/Automattic/wp-calypso/issues/87219#top but all the hamburger menus are displaying on my eiOS mobile device.


📌 Message to Author

@mgozdis Thanks for the report!
I can reproduce this in the Editor but not on the live site.

Here is how this looks on my iPhone:
![Captura de ecrã 2024-02-06 à(s) 16 53 01](https://github.com/Automattic/wp-calypso/assets/2540430/27509662-dddb-470a-b5c1-73b4b1b73b99)

I wonder if you can still reproduce this issue?
Thank you!

**Comment by mgozdis** [comment]
@filipanoscampos Yes, I can still reproduce on the live site and within the editor on a Simple site. I did not check the user's site reported in interaction as they still have Gutenberg deactivated as a workaround.

It seems to only be reproducible on a physical iOS device. I could not reproduce in emulators or scaling the browser window when viewing the live site.

**Comment by filipanoscampos** [comment]
Thanks!
I tested this on my own physical iPhone.
Do you have a URL I can check?

**Comment by mgozdis** [comment]
@filipanoscampos This was my test Simple site that does not display the mobile menu for me. I am testing on a really old iPhone though, but user in my interaction was using an iPhone 13 Pro and seeing the same unless deactivating Gutenberg: https://mgozdis25.wordpress.com/

With Gutenberg deactivated on the user's site, the overlay menu displays on my old iPhone, but it doesn't open, which is similar to other old bug reports I've seen and likely just not compatible with a very old iPhone.

**Comment by jp-imagines** [comment]
Ran into another report of this in 7706095-zd-a8c

* Menu doesn't reduce to the overlay style in the editor when using the mobile preview option. User and I replicated on Macbooks, and user also replicated from a Windows PC.
* On the live site itself, the mobile overlay menu does display (including on my Android phone, another HE's iPhone, and on my Macbook either by reducing the browser window width or using a mobile inspector preview).

**Comment by rickmgithub** [comment]
WHAT I FOUND:
- I can replicate the mobile menu not appearing in mobile preview in the editor but it appears on iOS devices for me

WHAT I DID:
On the basis that it's not working in the editor preview this in itself is a bug so marking as such and marking as triaged

**Comment by filipanoscampos** [comment]
Thank you both so much, I was able to reproduce this on an iPhone 7 with iOS 15.8 on Safari and Chrome.

📌 Update
Closed the other report in favor of this one.

**Comment by cuemarie** [comment]
Encountered here as well: 7711331-zen

**Comment by mgozdis** [comment]
Additional report here: 7712442-zd-a8c
User deactivated Gutenberg as a workaround since it was not working on frontend with physical device. They wouldn't specify iPhone model or iOS version.

**Comment by pauljacobson** [comment]
Adding another instance of this issue: 7686541-zen

**Comment by mgozdis** [comment]
Another report here 7738849-zd-a8c

**Comment by renata-franco** [comment]
Got a chat with a user having the same issue: 7742664-zen.

**Comment by mgozdis** [comment]
Additional report here with `Assembler` theme: 7754685-zd-a8c

**Comment by lsl** [comment]
Not able to replicate this (no iOS device handy), please pick this up for debug if you can. 

- There have been similar core issues in the past: https://github.com/WordPress/gutenberg/issues/36670. 
- Navigation block code here https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/navigation/edit/index.js. 
- It doesn't look like there would be any styling in Hevor theme causing this fbhepr%2Skers%2Sjcpbz%2Qcerzvhz%2Qgurzrf%2Suribe%2Sfglyr.pff%3Se%3Q1227o6r1-og
- It's possible we just need to update the theme template with newer block output fbhepr%2Skers%2Sjcpbz%2Qcerzvhz%2Qgurzrf%2Suribe%2Scnegf%2Surnqre.ugzy%3Se%3Q6ppopnsq%2328-og

**Comment by sbathompson-he** [comment]
This seems similar and is only happening on this user's mobile view: 8605069-zd-a8c

Here are the videos they supplied: 

https://github.com/user-attachments/assets/33744643-6e6e-4406-b0e1-ee0d45464d63
![IMG_1039](https://github.com/user-attachments/assets/daad3bb8-ed94-4a37-8a97-4ddb1cbe948a)

**Comment by miksansegundo** [comment]
There are different bugs being reported in this issue.

**The good news is none of them are reproducible using the latest Gutenberg version with the latest Safari version.** 

Tested with:
- Safari 17.2 and 17.5
- Gutenberg 18.9.0 and 19.0.0
- Themes Hevor and Assembler

These bugs are browser-specific, not theme-specific. If some are still reproducible, please check the browser or operating system version.

For optimal viewing and security, WordPress.com recommends using an up-to-date version, which generally means the latest two versions. See https://wordpress.com/support/browser-issues/#supported-browsers

Unfortunately, old Safari browser versions could still have these bugs. See the description in https://github.com/Automattic/wp-calypso/issues/87290#issue-2124329500

Note that Safari browser is tied to the iOS or MacOs versions. Meaning that users must update their device operating system to get the up-to-date Safari version. It's also unfortunate that Apple does not allow some very old macBooks and iPhones to be updated for performance and incompatibility reasons.

## Bugs

1. **Collapsed navigation (aka hamburger menu) not displayed in the editor mobile preview and the live site, which causes a borked layout.** 

Not reproduced with latest versions.

### BEFORE

|**Editor mobile preview on Safari desktop**|**Live site on Safari mobile**|
|-|-|
|![image](https://github.com/user-attachments/assets/ef9f3233-4fa9-4f9d-b952-906af94cf940)|![image](https://github.com/user-attachments/assets/f93c161b-7c26-4cc2-9ab0-ea7a4ee2e59e)|

### AFTER

|**Editor mobile preview on Safari desktop**|**Live site on Safari mobile**|
|-|-|
|<img width="1728" alt="Screenshot 2567-08-16 at 14 41 37" src="https://github.com/user-attachments/assets/ce46ed54-e2b8-4129-b237-f2f202c8000a">|![Simulator Screenshot - iPhone 15 - 2024-08-16 at 14 42 45](https://github.com/user-attachments/assets/3e0ab1d1-334e-4729-bf0d-6025bfabb65a)|


3. **Collapsed navigation button not responsive to clicks or touch events**

Not reproduced with latest versions.

|**Editor mobile preview on Safari desktop**|**Live site on Safari desktop**|**Live site on Safari mobile**|
|-|-|-|
|<img width="1728" alt="Screenshot 2567-08-16 at 14 41 50" src="https://github.com/user-attachments/assets/abc4ad54-bab4-4538-8f06-f882aecc87f5">|<img width="1728" alt="Screenshot 2567-08-16 at 16 18 53" src="https://github.com/user-attachments/assets/72f97063-14f9-4a0b-85b8-d11582d909e0">|![Simulator Screenshot - iPhone 15 - 2024-08-16 at 14 42 50](https://github.com/user-attachments/assets/376ea2a0-5528-463a-932e-23f4aa724df6)|

Please, feel free ask me questions and request help debugging similar issues. 

I believe we can lower the priority as in https://github.com/Automattic/wp-calypso/issues/87290#issuecomment-2222211816 cc: @inaikem

**Comment by nazilya06** [comment]
Another report: 8783752-zd-a8c
Twenty Twenty-Four theme

**Comment by mmtr** [comment]
> The good news is none of them are reproducible using the latest Gutenberg version with the latest Safari version.

I'll close this issue then.
